spec/services/user_mail_data_processor_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rspec/mocks'
+require './app/services/user_mail_data_processor.rb'
+
+RSpec.describe UserMailDataProcessor, type: :service do
+  subject(:data_processor) { UserMailDataProcessor.new }
+  let(:user_double) { double(user_name: 'test_user') }
+  let(:data_delimiter) { UserMailDataProcessor::DEFAULT_DATA_DELIMITER}
+
+  describe "#get_encrypted_payload" do
+    subject(:encrypted_payload) { data_processor.get_encrypted_payload(user_double) }
+    let(:decrypted_payload) { "#{user_double.user_name}#{data_delimiter}"}
+
+    it "returns encrypted payload" do
+      expect(encrypted_payload).not_to include decrypted_payload
```

Comment:
Here actually you can stub/mock the encryptor under the hood if you want to and return the string that you want to use it the test case;